04.class/memo_processing.js

```diff
@@ -0,0 +1,146 @@
+import * as readline from "node:readline/promises";
+import crypto from "node:crypto";
+import enquirer from "enquirer";
+const { prompt } = enquirer;
+import { FileOperation } from "./file_operation.js";
+import { QuestionsBuilder } from "./questions_builder.js";
+
+export class MemoProcessing {
+  constructor(destinationFile, options) {
+    this.destinationFile = destinationFile;
+    this.options = options;
+  }
+
+  async addNewMemo() {
+    try {
+      const file = new FileOperation(this.destinationFile);
+
+      let lines = [];
+      const newMemoContent = [];
+      const rl = readline.createInterface(process.stdin);
+      rl.on("line", (line) => {
+        lines.push(line);
+      });
+      rl.on("close", () => {
+        lines = lines.join("\n");
+        newMemoContent.push(lines);
+        const memoId = crypto.randomUUID();
+        const newMemoObj = { id: memoId, content: newMemoContent[0] };
+        this.#saveMemos(newMemoObj, file);
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async #saveMemos(newMemoObj, file) {
+    try {
+      const allMemos = await this.#readAllMemos();
+      allMemos.memos.push(newMemoObj);
+      file.writeTarget = allMemos;
+      file.write();
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async displayFirsLine() {
```

Comment:
`First` に typoがありそうですね。また、複数形(もしくは eachFirstLineなど)でもよさそうです。

```suggestion
  async displayFirstLines() {
```

Reply:
失礼いたしました..🙇‍♀️
typo箇所を修正し、複数形に変更いたしました！
ご提案までありがとうございます🙏